### PR TITLE
Add support for Policy

### PR DESCRIPTION
**Pull request by harkiratsm**
**Notes for Reviewers**

This PR fixes #

Added support to query and register policy.
 
New endpoints - 
```/api/meshmodel/policy/register```  ```POST```
```/api/meshmodel/policy``` ```GET```
```/api/meshmodel/model/{model}/policy``` ```GET```
```/api/meshmodel/model/{model}/policy/{name}``` ```GET```

- [ ] Document it under meshery API Endpoints Spreadsheet.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,6 +10,7 @@ replace (
 	github.com/go-openapi/strfmt => github.com/go-openapi/strfmt v0.20.2
 	github.com/kubernetes/kompose => github.com/meshery/kompose v1.0.1 // do not upgrade until https://github.com/meshery/kompose/pull/3 has been merged
 	github.com/kudobuilder/kuttl => github.com/layer5io/kuttl v0.4.1-0.20200806180306-b7e46afd657f
+	github.com/layer5io/meshkit => /home/harkiratsm/meshkit
```
Comment (leecalcote) [inline go.mod:13]: Will need to remove prior to merge.

Comment (harkiratsm) [reply go.mod:13]: ok

Comment (Revolyssup) [reply go.mod:13]: @harkiratsm Please remove it

File: server/handlers/policies_handlers.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	"github.com/layer5io/meshkit/models/meshmodel"
+	"github.com/layer5io/meshkit/models/meshmodel/core/types"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+)
+
+func (h *Handler) GetAllMeshmodelPolicy(rw http.ResponseWriter, r *http.Request) {
+	rw.Header().Add("Content-Type", "application/json")
+	enc := json.NewEncoder(rw)
+	typ := mux.Vars(r)["model"]
+	name := mux.Vars(r)["name"]
```
Comment (Revolyssup) [inline server/handlers/policies_handlers.go:22]: where are you getting the name from if this handler is commented? https://github.com/meshery/meshery/pull/7064/files#diff-c9f0fd2985523c81296d5a3b0b745ef02fa9d539a883a799977e0f4cb73f45e1R165

File: server/handlers/policies_handlers.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	"path"
+	"path/filepath"
+
+	"github.com/gorilla/mux"
+	"github.com/layer5io/meshkit/models/meshmodel"
+	"github.com/layer5io/meshkit/models/meshmodel/core/types"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+)
+
+func (h *Handler) GetAllMeshmodelPolicy(rw http.ResponseWriter, r *http.Request) {
```
Comment (Revolyssup) [inline server/handlers/policies_handlers.go:18]: Please add comments to generate swagger docs for this endpoint. Refer to component_handler.go

File: server/handlers/policies_handlers.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	rw.Header().Add("Content-Type", "application/json")
+	enc := json.NewEncoder(rw)
+	typ := mux.Vars(r)["model"]
+	name := mux.Vars(r)["name"]
+
+	res := h.registryManager.GetEntities(&v1alpha1.PolicyFilter{
+		Kind:      name,
+		ModelName: typ,
+	})
+	unique := make(map[string]bool)
```
Comment (Revolyssup) [inline server/handlers/policies_handlers.go:28]: @harkiratsm Why are you not allowing multiple policies with same model name?

File: server/handlers/policies_handlers.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+				return err
+			}
+			rs = append(rs, p)
+		}
+		return nil
+	})
+	return
+}
+
+func RegisterStaticMeshmodelPolicy(rm meshmodel.RegistryManager, sourceDirPath string) (err error) {
```
Comment (Revolyssup) [inline server/handlers/policies_handlers.go:93]: What is the purpose of this handler?

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: server/handlers/policies_handlers.go (hunk 5 above)
Comment (harkiratsm) [reply server/handlers/policies_handlers.go:93]: To seed static policy defined under ```/meshmodel/policies``` folder.

File: server/handlers/policies_handlers.go (hunk 4 above)
Comment (harkiratsm) [reply server/handlers/policies_handlers.go:28]: At first, I wanted to remove the duplicate policy, so this function gives us unique policy.

Comment (harkiratsm) [reply server/handlers/policies_handlers.go:28]: Now considering your thoughts, I will change this behavior.

File: server/handlers/policies_handlers.go (hunk 2 above)
Comment (harkiratsm) [reply server/handlers/policies_handlers.go:22]: fixed.

File: server/handlers/policies_handlers.go (hunk 3 above)
Comment (harkiratsm) [reply server/handlers/policies_handlers.go]: Done

**Comment by l5io** [comment]
Task linked: [CU-85zrj2fre custom validation rules to be included in a model package](https://app.clickup.com/t/85zrj2fre)

File: server/handlers/policies_handlers.go (hunk 5 above)
Comment (Revolyssup) [reply server/handlers/policies_handlers.go:121]: But why is this in handlers package? Any seeding is not done over HTTP

**Review by leecalcote — CHANGES_REQUESTED** [review]
@harkiratsm there are a number of build failures here.

I'll kick off a new set of builds...

File: server/cmd/main.go [hunk 6]
```diff
@@ -295,6 +295,19 @@ func main() {
 		}
 		go hc.MeshModelSummaryChannel.Publish()
 	}()
+	// seed policys
+	go func() {
+		staticPolicyPath, err := filepath.Abs("../meshmodel/policies")
+		if err != nil {
+			fmt.Println("Error registering policies: ", err.Error())
```
Comment (leecalcote) [inline server/cmd/main.go:302]: @Revolyssup Println is not appropriate, is it?

File: server/handlers/policies_handlers.go (hunk 5 above)
Comment (harkiratsm) [reply server/handlers/policies_handlers.go:121]: If it deviates from the purpose of handlers, I will make the appropriate changes to the policy and relationship seeding.

Comment (Revolyssup) [reply server/handlers/policies_handlers.go:121]: what? This is not a handler.

Comment (Revolyssup) [reply server/handlers/policies_handlers.go:121]: Refer to how static components are registered please

Comment (harkiratsm) [reply server/handlers/policies_handlers.go]: Done

Comment (harkiratsm) [reply server/handlers/policies_handlers.go]: Fixed the static registering of relationships as well.

**Comment by harkiratsm** [comment]
https://github.com/meshery/meshkit/pull/272

**Comment by harkiratsm** [comment]
All the tests are passing.

File: server/meshmodel/policies/validator.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
+    "kind":"Validation",
+    "subType":"rjsf",
+    "apiVersion":"core.meshery.io/v1alpha1",
+    "model":{
+        "name":"kubernetes"
```
Comment (Revolyssup) [inline server/meshmodel/policies/validator.json:6]: @harkiratsm Add version and other fields.

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: server/meshmodel/policies/validator.json (hunk 7 above)
Comment (harkiratsm) [reply server/meshmodel/policies/validator.json]: Done :heavy_check_mark:

**Pull request closed without merging**